File: cpp/topKFrequent.cpp
#include <algorithm>
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>

using std::vector;
using std::unordered_map;

class Solution {
public:
    static vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map <int, int> temp;
        for (auto i : nums){
            temp[i]++;
        }
        vector<std::pair<int, int>> tempV;
        for (std::pair<int, int> i : temp){
            tempV.push_back(i);
        }
        std::sort(tempV.rbegin(), tempV.rend(), [](std::pair<int, int> &left, std::pair<int, int> &right){
            return left.second < right.second;
        });
        vector<int> result;
        for (int i = 0; i < k; i++){
            result.push_back(tempV[i].first);
        }
        return result;
    }
};

int main(){
    vector<int> test1 = {1,1,1,2,2,3};
    vector<int> result = Solution::topKFrequent(test1, 2);
    for (int i : result){
        std::cout << i << std::endl;
    }
}

File: python/topKFrequent.py
from typing import List
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        freq = [[] for i in (range(len(nums)+1))]
        for n in nums:
            count[n] = 1 + count.get(n, 0);
        for n, c in count.items():
            freq[c].append(n)
        result = []
        for i in range(len(freq) - 1, 0, -1):
            for n in freq[i]:
                result.append(n)
            if (len(result) == k):
                return result
            
if __name__ == "__main__":
    nums = [1,1,1,2,2,3]
    k = 2
    result = Solution().topKFrequent(nums, k)
    print(result)